crates/oxc_transformer/src/es2022/class_properties/supers.rs

```diff
@@ -0,0 +1,76 @@
+//! ES2022: Class Properties
+//! Transform of super member expressions.
+
+use oxc_ast::ast::*;
+use oxc_span::GetSpan;
+use oxc_traverse::TraverseCtx;
+
+use crate::Helper;
+
+use super::ClassProperties;
+
+impl<'a, 'ctx> ClassProperties<'a, 'ctx> {
+    /// Transform member expression where object is a super.
+    ///
+    /// - `super.prop` -> `_superPropGet(_classBinding, "prop", _classBinding)`
+    /// - `super[expr]` -> `_superPropGet(_classBinding, expr, _classBinding)`
+    //
+    // `#[inline]` so that compiler sees that `expr` is an `Expression::StaticMemberExpression`
+    // or `Expression::ComputedMemberExpression`.
+    #[inline]
+    pub fn transform_member_expression(
+        &mut self,
+        expr: &mut Expression<'a>,
+        ctx: &mut TraverseCtx<'a>,
+    ) {
+        let member = expr.to_member_expression();
+        if matches!(member.object(), Expression::Super(_)) {
+            self.transform_member_expression_impl(expr, ctx);
+        }
+    }
+
+    fn transform_member_expression_impl(
+        &mut self,
+        expr: &mut Expression<'a>,
+        ctx: &mut TraverseCtx<'a>,
+    ) {
+        let property = match expr {
+            // `super.prop` -> `"prop"`
+            Expression::StaticMemberExpression(member) => {
+                let property = &member.property;
+                ctx.ast.expression_string_literal(
+                    property.span,
+                    property.name.clone(),
+                    Some(property.name.clone()),
+                )
+            }
+            // `super[expr]` -> `expr`
+            Expression::ComputedMemberExpression(member) => {
+                ctx.ast.move_expression(&mut member.expression)
+            }
+            Expression::PrivateFieldExpression(_) => {
+                unreachable!("`super` cannot access private fields")
+            }
+            _ => return,
+        };
+
+        *expr = self.create_super_prop_get(expr.span(), property, ctx);
+    }
+
+    // `_superPropGet(_classBinding, "prop", _classBinding)`
+    fn create_super_prop_get(
+        &mut self,
+        span: Span,
+        property: Expression<'a>,
+        ctx: &mut TraverseCtx<'a>,
+    ) -> Expression<'a> {
+        let class_binding = self.class_bindings.get_or_init_temp_binding(ctx);
```

Comment:
This needs to use `ClassProperties::get_temp_binding` instead (defined in `static_prop_init.rs`, you'll need to make that method `pub(super)`).

```suggestion
        let class_binding = self.get_temp_binding(ctx);
```

It's really annoying we have to do this, and it's not at all obvious. The problem is that we store `class_bindings` in 2 places and which is the "source of truth" depends on whether the class has private properties or not. This is very bad design (my fault), but I couldn't find a better way at present.